cpp/cmake/tosca_coverage.cmake

```diff
@@ -1,36 +1,60 @@
 include_guard(GLOBAL)
 
-
 option(TOSCA_COVERAGE "Enable coverage report for evmzero." OFF)
 
 if(TOSCA_COVERAGE)
-  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+ 
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```

Comment:
The old version checking explicitly for the GNU compiler seems to be the more accurate condition for hat is reported by the error message below. So why would this condition be changed?

Reply:
yes, this has been reverted.